zebra-network/src/peer_set/initialize.rs

```diff
@@ -211,7 +211,18 @@ where
     S: Service<(TcpStream, SocketAddr), Response = peer::Client, Error = BoxError> + Clone,
     S::Future: Send + 'static,
 {
-    let listener = TcpListener::bind(addr).await?;
+    let check_listener = TcpListener::bind(addr).await;
+
+    let listener = match check_listener {
+        Ok(l) => l,
+        Err(_) => panic!(
+            "{} {} {}",
+            "Listener failed. Port already in use.",
+            "Is another instance of zebrad or zcashd running in your local machine?",
+            "Consider changing the default port in the configuration file."
+        ),
```

Comment:
1. We can't assume that the error is "port already in use"
2. Rust has support for multi-line string literals: https://stackoverflow.com/questions/29483365/what-is-the-syntax-for-a-multiline-string-literal/29483453#29483453

```suggestion
        Err(e) => panic!(
            "Opening Zcash network protocol listener failed: {:?}. \
             Hint: Is another instance of zebrad or zcashd running in your local machine? \
             Consider changing the default port in the configuration file.",
             e,
        ),
```